src/controllers/expenseController.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const userServices = require('../services/userServices');
+const expenseServices = require('../services/expenseServices');
+
+function getAllExpenses(req, res) {
+  const {
+    userId,
+    categories,
+    from,
+    to,
+  } = req.query;
+
+  const filteredExpenses = expenseServices
+    .getExpenses(userId, categories, from, to);
+
+  res.send(filteredExpenses);
+}
+
+function getExpenseByUserId(req, res) {
+  const { userId } = req.params;
+
+  if (!userId) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const foundExpense = expenseServices.getExpenseByUserId(userId);
+
+  if (!foundExpense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.statusCode = 200;
+  res.send(foundExpense);
+}
+
+function createExpense(req, res) {
+  const {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  } = req.body;
+
+  const foundUser = userServices.getByUserId(userId);
+
+  if (!foundUser) {
+    res.sendStatus(400);
```

Comment:
Return status code 404 instead of 400 if foundUser is not found, as the requirement states 'Return 404 if expected entity doesn't exist'.
```suggestion
    res.sendStatus(404);
```